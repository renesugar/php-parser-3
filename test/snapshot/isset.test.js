const parser = require('../main');

describe("isset", function() {
  it("simple", function() {
    expect(parser.parseEval('isset($var);')).toMatchSnapshot();
  });
  it("multiple", function() {
    expect(parser.parseEval('isset($var, $var, $var);')).toMatchSnapshot();
  });
  it("assign", function() {
    expect(parser.parseEval('$var = isset($var);')).toMatchSnapshot();
  });
});
