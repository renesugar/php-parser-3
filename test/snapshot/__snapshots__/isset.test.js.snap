// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isset assign 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "var",
        },
        "operator": "=",
        "right": Isset {
          "kind": "isset",
          "variables": Array [
            Variable {
              "byref": false,
              "curly": false,
              "kind": "variable",
              "name": "var",
            },
          ],
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`isset multiple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Isset {
        "kind": "isset",
        "variables": Array [
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "var",
          },
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "var",
          },
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "var",
          },
        ],
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`isset simple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Isset {
        "kind": "isset",
        "variables": Array [
          Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "var",
          },
        ],
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
